receiver/prometheusreceiver/factory.go

```diff
@@ -45,6 +46,9 @@ func NewFactory() component.ReceiverFactory {
 func createDefaultConfig() component.Config {
 	return &Config{
 		ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr)),
+		PrometheusConfig: &promconfig.Config{
+			GlobalConfig: promconfig.DefaultGlobalConfig,
+		},
```

Comment:
```suggestion
		PrometheusConfig: &promconfig.DefaultConfig
```

Reply:
@dashpole 
suggested change will fail the tests because the struct `promconfig.DefaultConfig` directed by the pointer will be shared through the test (and it seems to be modified in [here](https://github.com/ymtdzzz/opentelemetry-collector-contrib/blob/3d0d925f1722fc5ded544bd02953c4fbd564b6bd/receiver/prometheusreceiver/config.go#L280) ).

test fail logs (with debug print)
```
$ go test -race -timeout 30
0s --tags="" -run 'TestRejectUnsupportedPrometheusFeatures|TestNonExistentAuthCredentialsFile' ./... -v
=== RUN   TestRejectUnsupportedPrometheusFeatures
--- PASS: TestRejectUnsupportedPrometheusFeatures (0.01s)
=== RUN   TestNonExistentAuthCredentialsFile
want(prefix): error checking authorization credentials file "/nonexistentauthcredentialsfile"
got: unsupported features:
	alert_config.alertmanagers
	alert_config.relabel_configs
	remote_read
	remote_write
	rule_files
    config_test.go:188: 
        	Error Trace:	/home/ymtdzzz/development/github.com/ymtdzzz/opentelemetry-collector-contrib/receiver/prometheusreceiver/config_test.go:188
        	Error:      	Should be true
        	Test:       	TestNonExistentAuthCredentialsFile
--- FAIL: TestNonExistentAuthCredentialsFile (0.00s)
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver	0.109s
testing: warning: no tests to run
PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal	(cached) [no tests to run]
FAIL
```

to pass the tests (and avoid the risk of conflicts), we should create new one in every `createDefaultConfig()` calls.

```golang
	return &Config{
		ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr)),
		PrometheusConfig: &promconfig.Config{
			GlobalConfig: promconfig.DefaultGlobalConfig,
		},
	}
// Or
	pdcfg := promconfig.DefaultConfig
	return &Config{
		ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr)),
		PrometheusConfig: &pdcfg,
	}
```

Reply:
Does the same apply to the current code if someone overrides `GlobalConfig` in their config?

Reply:
@dashpole 

> if someone overrides `GlobalConfig` in their config?

In that case, only parameters specified in their config will be overridden but `promconfig.DefaultGlobalConfig` itself will not be modified because it's not a pointer.
In fact, `Unmarshal()` is called only once at startup, so this "conflict" seems to be a test-only issue.

config:
```yaml
receivers:
  prometheus:
    config:
      global:
        scrape_interval: 10m
      scrape_configs:
        - job_name: 'otel-collector'
          scrape_interval: 5s
          static_configs:
            - targets: ['0.0.0.0:8888']
```

result of `Unmarshal()`:

```
otel-col  | ======= before unmarshall =======
otel-col  | config: &{{{{prometheus } <nil>}} global:
otel-col  |   scrape_interval: 1m
otel-col  |   scrape_timeout: 10s
otel-col  |   evaluation_interval: 1m
otel-col  |  0s 0 false  <nil> <nil>}
otel-col  | -----------
otel-col  | promconfig.DefaultGlobalConfig: global:
otel-col  |   scrape_interval: 1m
otel-col  |   scrape_timeout: 10s
otel-col  |   evaluation_interval: 1m

otel-col  | ======= after unmarshall =======
otel-col  | config: &{{{{prometheus } <nil>}} global:
otel-col  |   scrape_interval: 10m // overridden by config
otel-col  |   scrape_timeout: 10s
otel-col  |   evaluation_interval: 1m
otel-col  | scrape_configs:
otel-col  | - job_name: otel-collector
otel-col  |   honor_timestamps: true
otel-col  |   scrape_interval: 5s
otel-col  |   scrape_timeout: 5s
otel-col  |   metrics_path: /metrics
otel-col  |   scheme: http
otel-col  |   follow_redirects: true
otel-col  |   enable_http2: true
otel-col  |   static_configs:
otel-col  |   - targets:
otel-col  |     - 0.0.0.0:8888
otel-col  |  0s 0 false  <nil> map[global:map[scrape_interval:10m] scrape_configs:[map[job_name:otel-collector scrape_interval:5s static_configs:[map[targets:[0.0.0.0:8888]]]]]]}
otel-col  | -----------
otel-col  | promconfig.DefaultGlobalConfig: global: // not modified
otel-col  |   scrape_interval: 1m
otel-col  |   scrape_timeout: 10s
otel-col  |   evaluation_interval: 1m
```

Reply:
Thanks for checking.